controllers/resource_helper.go

```diff
@@ -170,22 +181,59 @@ func buildContainerSpec(ctx context.Context, r *LlamaStackDistributionReconciler
 
 // resolveContainerResources ensures the container always has CPU and memory
 // requests defined so that HPAs using utilization metrics can function.
-func resolveContainerResources(spec llamav1alpha1.ContainerSpec) corev1.ResourceRequirements {
+func resolveContainerResources(spec llamav1alpha1.ContainerSpec, workers int32, workersSet bool) corev1.ResourceRequirements {
 	resources := spec.Resources
 
+	ensureRequests(&resources, workers)
+	if workersSet {
+		ensureLimitsMatchRequests(&resources)
+	}
+
+	cpuReq := resources.Requests[corev1.ResourceCPU]
+	memReq := resources.Requests[corev1.ResourceMemory]
+	cpuLimit := resources.Limits[corev1.ResourceCPU]
+	memLimit := resources.Limits[corev1.ResourceMemory]
+
+	ctrlLog.Log.WithName("resource_helper").WithValues(
+		"workers", workers,
+		"workersEnabled", workersSet,
+	).V(1).Info("Defaulted resource values for llama-stack container",
+		"cpuRequest", cpuReq.String(),
+		"memoryRequest", memReq.String(),
+		"cpuLimit", cpuLimit.String(),
+		"memoryLimit", memLimit.String(),
+	)
+
+	return resources
+}
+
+func ensureRequests(resources *corev1.ResourceRequirements, workers int32) {
 	if resources.Requests == nil {
 		resources.Requests = corev1.ResourceList{}
 	}
 
 	if cpuQty, ok := resources.Requests[corev1.ResourceCPU]; !ok || cpuQty.IsZero() {
-		resources.Requests[corev1.ResourceCPU] = llamav1alpha1.DefaultServerCPURequest
+		// Default to 1 full core per worker unless user overrides.
+		resources.Requests[corev1.ResourceCPU] = resource.MustParse(strconv.Itoa(int(workers)))
```

Comment:
This has me concerned that we'll block X CPU's on individual nodes
if workers=4 , then we'll reserve 4 CPU's that can go unused in the cluster even if the workload is IO bound or idle

I think it might be better to set the limit like this but keep the request a lot lower, but given the time constraints I'm ok if you want to keep it like this for now and we can refine later.